### PR TITLE
create institution API

### PR DESCRIPTION
**Pull request by antonioT90**
CreateInstitution API
onboardingInstitution will not more create institution

### CODE REVIEW

File: src/main/resources/interface-specification.yml [hunk 1]
```diff
@@ -597,6 +597,40 @@ paths:
             application/problem+json:
               schema:
                 $ref: '#/components/schemas/Problem'
+  '/institutions/{externalId}':
```
Comment (beetlecrunch) [inline src/main/resources/interface-specification.yml:600]: `externalId` is already part of the request body, why this path parameter?
You are trying to create a resource that may not exist yet.

Comment (antonioT90) [reply src/main/resources/interface-specification.yml:600]: @beetlecrunch there is not a request body in this API :)

File: src/main/resources/interface-specification.yml [hunk 2]
```diff
@@ -597,6 +597,40 @@ paths:
...
+    post:
+      security: [ { } ]
```
Comment (galales) [inline src/main/resources/interface-specification.yml]: probably you already did your considerations, but seeing this I have to say it 😄 

Not sure if this service is under an authenticated layer, but not having authentication on a "write" route can lead to issues

File: src/main/scala/it/pagopa/interop/partyprocess/api/impl/ProcessApiServiceImpl.scala [hunk 3]
```diff
@@ -154,19 +154,24 @@ class ProcessApiServiceImpl(
...
-            .orElse(Option(institution))
+    institutionId
+      .fold {
+        institutionExternalId.traverse(externalId =>
+          partyManagementService.retrieveInstitutionByExternalId(externalId)(bearer)
         )
-    }
+      } { institutionId =>
+        partyManagementService
+          .retrieveInstitution(UUID.fromString(institutionId))(bearer)
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/partyprocess/api/impl/ProcessApiServiceImpl.scala]: `UUID.fromString` throws an exception if the given string is not a uuid

File: src/main/resources/interface-specification.yml (hunk 2 above)
Comment (antonioT90) [reply src/main/resources/interface-specification.yml]: you're right :) It was not clear to me this line :D now I fixed it ;)

File: src/main/scala/it/pagopa/interop/partyprocess/api/impl/ProcessApiServiceImpl.scala (hunk 3 above)
Comment (antonioT90) [reply src/main/scala/it/pagopa/interop/partyprocess/api/impl/ProcessApiServiceImpl.scala]: the institutionId is supposed to be a UIID:
fixed getInstitutionByOptionIdAndOptionExternalId method firm

File: src/main/scala/it/pagopa/interop/partyprocess/api/impl/ProcessApiServiceImpl.scala [hunk 4]
```diff
@@ -117,10 +117,11 @@ class ProcessApiServiceImpl(
...
 
     val result: Future[OnboardingInfo] = for {
-      bearer           <- getFutureBearer(contexts)
-      uid              <- getUidFuture(contexts)
-      userId           <- uid.toFutureUUID
-      institution      <- getInstitutionByOptionIdAndOptionExternalId(institutionId, institutionExternalId)(bearer)
+      bearer <- getFutureBearer(contexts)
+      uid    <- getUidFuture(contexts)
+      userId <- uid.toFutureUUID
+      institutionUuid = institutionId.map(UUID.fromString)
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/partyprocess/api/impl/ProcessApiServiceImpl.scala]:
```suggestion
      institutionUuid <- institutionId.traverse(_.toFutureUUID)
```

**Review by galales — APPROVED** [review]

**Review by beetlecrunch — APPROVED** [review]

**Pull request merged**